waltz-tools/src/test/java/com/wepay/waltz/tools/zk/ZooKeeperCliTest.java

```diff
@@ -259,6 +259,50 @@ public void testAddExistingStorageNodeShouldFail() throws Exception {
         }
     }
 
+    @Test
+    public void testAddMisformattedStorageNodeShouldFail() throws Exception {
+        // set up streams
+        PrintStream originalErr = System.err;
+        ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+        System.setErr(new PrintStream(errContent, false, "UTF-8"));
+
+        ZooKeeperServerRunner zooKeeperServerRunner = new ZooKeeperServerRunner(0);
+        String connectString = zooKeeperServerRunner.start();
+        String configFilePath = IntegrationTestHelper.createYamlConfigFile(DIR_NAME, CONFIG_FILE_NAME, createProperties(connectString, CLUSTER_ROOT));
+
+        try {
+            ZNode clusterRootZNode = new ZNode(CLUSTER_ROOT);
+            ZNode storeRoot = new ZNode(clusterRootZNode, StoreMetadata.STORE_ZNODE_NAME);
+            ZNode groupNode = new ZNode(storeRoot, StoreMetadata.GROUP_ZNODE_NAME);
+            ZooKeeperClient zkClient = new ZooKeeperClientImpl(connectString, ZK_SESSION_TIMEOUT);
+            ZooKeeperCli.Create.createCluster(zkClient, clusterRootZNode, CLUSTER_NAME, NUM_PARTITIONS);
+            ZooKeeperCli.Create.createStores(zkClient, clusterRootZNode, NUM_PARTITIONS, STORAGE_GROUP_MAP, STORAGE_CONNECTION_MAP);
+
+            int prevGroupSize = zkClient.getData(groupNode, GroupDescriptorSerializer.INSTANCE).value.groups.size();
+
+            // add mis-formatted storage node should fail
```

Comment:
may be you can just add the below code to the `testAddStorageNode()` itself.

Reply:
Actually, get a second opinion on that. Looks like each test case in ZooKeeperCli is handling only one specific scenario. So, may be having a separate test case should be fine.

Reply:
> may be you can just add the below code to the `testAddStorageNode()` itself.
Yes that makes sense, but I was thinking that there's like testAddExistingStorageNodeShouldFail under ZookerperCliTest and those tests seem fine-grained so I still added a new unit test. But I'm unsure which is the better way to go..